### PR TITLE
Refactor handling of recent projects

### PR DESCRIPTION
**Pull request by ladvoc**
While reading through the codebase, I noticed quite a bit of duplicate code related to recent projects. Recent projects can be viewed and opened from multiple locations, and each location reimplements the same functionality (e.g. storing/loading from `UserDefaults`, moving the most recently opened project to the front of the list, etc.). I introduced `RecentProjectsStore`, accessible through a global singleton, to encapsulate this functionality. Now, each location dealing with recent projects does so through method calls to the `RecentProjectsStore`—eliminating the duplicate code.

### CODE REVIEW

**Comment by 0xWDG** [comment]
Thanks, Great work, I'll test it later but it looks promising 👍

File: AuroraEditor/Application/AppDelegate.swift [hunk 1]
```diff
@@ -38,6 +39,8 @@ final class AppDelegate: NSObject, NSApplicationDelegate, ObservableObject {
 
     private var updateModel: UpdateObservedModel = .shared
 
+    var cancellable = Set<AnyCancellable>()
```
Comment (nanashili) [inline AuroraEditor/Application/AppDelegate.swift:42]: Is the cancellable called?

**Comment by nanashili** [comment]
Oo, i like how this is being handled

**Review by 0xWDG — APPROVED** [review]

**Review by nanashili — APPROVED** [review]

File: AuroraEditor/Application/AppDelegate.swift (hunk 1 above)
Comment (ladvoc) [reply AuroraEditor/Application/AppDelegate.swift:42]: When subscribing to `RecentProjectsStore.shared.$paths` to construct the recents menu when the paths change, I store the canellable in this set.

**Comment by 0xWDG** [comment]
@ladvoc i'm sorry. i forgot to add you to the all-contributors.
Thanks for your PR ❤️.

Your name is now listed in the repo and our website https://auroraeditor.com/#/contributors

**Pull request merged**